src/naive/date/mod.rs

```diff
@@ -1999,7 +1993,7 @@ impl Iterator for NaiveDateWeeksIterator {
 
     fn next(&mut self) -> Option<Self::Item> {
         let current = self.value;
-        self.value = current.checked_add_days(Days::new(7))?;
+        self.value = current.checked_add_days(Days::new(7)).ok()?;
```

Comment:
Unrelated to this PR, so just ignore this comment.
I wonder if the `NaiveDateWeeksIterator` can cover the complete range of `NaiveDate`, or if it fails one week before the end.